nativelib/src/main/resources/scala-native/gc/immix/ThreadManager.c

```diff
@@ -0,0 +1,99 @@
+#include <setjmp.h>
+#include <signal.h>
+#include <stdio.h>
+
+#include "ThreadManager.h"
+#include "GCTypes.h"
+#include "semaphore/Semaphore.h"
+
+#define SIG_SUSPEND SIGXFSZ
+#define SIG_RESUME SIGXCPU
+
+// Signal handlers can't receive parameters.
```

Comment:
Yes, however we can replace usage of `sa_handler`  with `sa_sigaction`. We can pass pointer to struct containing semaphore and pointer for stack_top in `siginfo_t.si_ptr`, however we would need to replace `pthread_kill` with`pthread_sigqueue`. I'm not very experienced in C signals but such approach might remove this global  variables and  help to create easier to understand structure of thread manager workflow.